kinto/views/records.py

```diff
@@ -55,23 +53,32 @@ def process_record(self, new, old=None):
         if not schema or not asbool(settings.get(schema_validation)):
             return new
 
-        collection_timestamp = self._collection[self.model.modified_field]
-
+        # Remove internal and auto-assigned fields from schema and record.
+        internal_fields = (self.model.id_field,
+                           self.model.modified_field,
+                           self.schema_field,
+                           self.model.permissions_field)
+        required_fields = [f for f in schema.get('required', []) if f not in internal_fields]
+        if required_fields:
+            schema = {**schema, 'required': required_fields}
+        else:
+            schema = {f: v for f, v in new.items() if f != 'required'}
```

Comment:
This approach has some advantages:
* it avoids breaking with existing collection metadata
* we strip record and schema at the same place (stripping record fields will still be necessary if the users don't allow additional props on their schemas)
* we are in a resource class, where `self.model.id_field` etc. are available. It's not the case within the colander validator for collection schema
* it is fairly similar to what there were before this PR, so I'm not chocked by those changes

Please expose your whys :)